arm-appinsights/2015-05-01/swagger/component_API.json

```diff
@@ -0,0 +1,376 @@
+  "info": {
+    "title": "AppInsightsManagementClient",
+    "description": "Azure Application Insights client for Components.",
+    "version": "2015-05-01"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow.",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/providers/microsoft.insights/components": {
+      "get": {
+        "description": "Gets a list of all Application Insights components within a subscription.",
+        "operationId": "Component_List",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "A list containing 0 or more Application Insights component definitions.",
+            "schema": {
+              "$ref": "#/definitions/ApplicationInsightsComponentListResult"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "ComponentsList.json": { "$ref": "../examples/ComponentsList.json" }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/microsoft.insights/components": {
+      "get": {
+        "description": "Gets a list of Application Insights components within a resource group.",
+        "operationId": "Component_ListByResourceGroup",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "A list containing 0 or more Application Insights component definitions.",
+            "schema": {
+              "$ref": "#/definitions/ApplicationInsightsComponentListResult"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "ComponentListByResourceGroup": { "$ref": "../examples/ComponentListByResourceGroup.json" }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/microsoft.insights/components/{resourceName}": {
```

Comment:
- [x] **Done**. Added Patch operation and simplified tags-only resource for specifying it.